extensions.html

```diff
@@ -445,16 +507,122 @@ <h2>
+                  and normalized to the range [-1.0, 1.0] where -1.0 is the leftmost
+                  coordinate and 1.0 is the rightmost coordinate.</p>
+                <p>
+                <p>Possible implementation:</p>
+                <ul>
+                  <li>`normalizedX = (2.0 * touchData.x / surfaceDimensionX) - 1`</li>
+                </ul>
+                </p>
+              </li>
+              <li>
+                <p>Let |normalizedY:float| be a value calculated from the current
+                  touch input y coordinate in relationship to the surfaceDimensionY
+                  and normalized to the range [-1.0, 1.0] where -1.0 is the topmost
+                  coordinate and 1.0 is the bottommost coordinate.
+                </p>
+                <p>
+                <p>Possible implementation:</p>
+                <ul>
+                  <li>`normalizedY = (2.0 * touchData.y / surfaceDimensionY) - 1`</li>
+                </ul>
+                </p>
+              </li>
+              <li>Let |touchEvent:GamepadTouch| be a {{GamepadTouch}}.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/touchId}} to be
+              `touchData.touchID`.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/surfaceId}} to be
+              device surfaceId.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/position}}[0] to be
+              |normalizedX:float|.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/position}}[1] to be
+              |normalizedY:float|.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/surfaceDimensions}}[0] to be
+              device max x-axis coordinate.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/surfaceDimensions}}[1] to be
+              device max y-axis coordinate.<p>(<i>note: surfaceDimensions can be set to `null`</i>)</p>
+              </li>
+              <li>Add |touchEvent| to |gamepad|.{{Gamepad/[[touchEvents]]}}.
+              </li>
+            </ol>
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          Constructing a `Gamepad`
+        </h3>
+        <p>
+          This section supplements the <a href="https://w3c.github.io/gamepad/#constructing-a-gamepad">
+            Constructing a `Gamepad`</a> section of the main 
+          <a href="https://w3c.github.io/gamepad/">Gamepad specification</a>.
+        </p>
+        <p>
+          In addtion to the steps defined in the main 
+          <a href="https://w3c.github.io/gamepad/">Gamepad specification</a>.
+          When constructing <dfn>a new `Gamepad`</dfn> representing a connected gamepad device
+          perform the following steps:
+        </p>
+        <ol>
+          <li>Let |gamepad:Gamepad| be a newly created {{Gamepad}} instance:
+            <ol>
+              <li>Initialize |gamepad|'s {{Gamepad/touchEvents}} to the result of
```

Comment:
I think here we should initialize the touchEvents internal slot to an empty list if the gamepad has any touch surfaces, otherwise null. The [=updating touchEvents=] algorithm will be called automatically whenever the first touch inputs are received so we don't need to call it here.

The actual touchEvents attribute (not the internal slot) should define a getter algorithm that returns the data from the internal slot. That way the attribute is initialized when we initialize the internal slot. We use the internal slot because the attribute is readonly so technically it has no setter but we still need a way to update it.

Reply:
Updated.